aptos-move/framework/aptos-framework/sources/aggregator/optional_aggregator.move

```diff
@@ -164,6 +180,22 @@ module aptos_framework::optional_aggregator {
         }
     }
 
+    /// Subtracts `value` to optional aggregator, returns `true` on success and `false` on underflow.
+    public fun try_sub(optional_aggregator: &mut OptionalAggregator, value: u128): bool {
+        if (option::is_some(&optional_aggregator.aggregator)) {
+            let aggregator = option::borrow_mut(&mut optional_aggregator.aggregator);
+            aggregator::try_sub(aggregator, value)
+        } else {
+            let integer = option::borrow_mut(&mut optional_aggregator.integer);
+            if (value <= integer.value) {
```

Comment:
let's implement `try_sub` for integer instead and use it here.

Reply:
Also `sub` for integer (and `add`) would simply reuse these new `try_` methods

Reply:
Done.